src/Microsoft.Management.Deployment/PackageCatalog.cpp

```diff
@@ -107,70 +108,81 @@ namespace winrt::Microsoft::Management::Deployment::implementation
         return S_OK;
     }
 
+    winrt::Microsoft::Management::Deployment::FindPackagesResult GetFindPackagesResult(HRESULT hr, bool isTruncated, Windows::Foundation::Collections::IVector<Microsoft::Management::Deployment::MatchResult> matches)
+    {
+        auto findPackagesResult = winrt::make_self<wil::details::module_count_wrapper<
+            winrt::Microsoft::Management::Deployment::implementation::FindPackagesResult>>();
+        // TODO: Add search timeout and error code.
+        winrt::Microsoft::Management::Deployment::FindPackagesResultStatus status = FindPackagesResultStatus(hr);
+        findPackagesResult->Initialize(status, isTruncated, matches);
+        return *findPackagesResult;
+    }
+
     winrt::Microsoft::Management::Deployment::FindPackagesResult PackageCatalog::FindPackages(winrt::Microsoft::Management::Deployment::FindPackagesOptions const& options)
     {
         winrt::Microsoft::Management::Deployment::FindPackagesResultStatus::Ok;
         bool isTruncated = false;
         Windows::Foundation::Collections::IVector<Microsoft::Management::Deployment::MatchResult> matches{ winrt::single_threaded_vector<Microsoft::Management::Deployment::MatchResult>() };
         ::AppInstaller::Repository::SearchRequest searchRequest;
 
-        HRESULT hr = PopulateSearchRequest(&searchRequest, options);
-        if (SUCCEEDED(hr))
+        HRESULT hr = S_OK;
+        if (FAILED(hr = EnsureComCallerHasCapability()))

```

Comment:
Why is this the place to require the capability?  Why is it not the object factory method?

I would think that installing and other modifying APIs would require `packageManagement`, while searching would require `packageQuery`.  So maybe it does make sense to do the checks in the functions themselves.  But then we are requiring the wrong capabilities.

Reply:
Read the description! Fine, probably I should add a comment too. The ActivateInstance calls seem to be coming from a com service svchost, so the process id isn't useful for doing the check.